exp/api/v1alpha3/azuremanagedcontrolplane_types.go

```diff
@@ -77,6 +77,53 @@ type AzureManagedControlPlaneSpec struct {
 	// +kubebuilder:validation:Enum=Basic;Standard
 	// +optional
 	LoadBalancerSKU *string `json:"loadBalancerSKU,omitempty"`
+
+	// AadProfile is Azure Active Directory configuration to integrate with AKS for aad authentication.
+	// +optional
+	AADProfile *AADProfile `json:"aadProfile,omitempty"`
+}
+
+// AADProfile is Azure Active Directory configuration to integrate with AKS for aad authentication.
+type AADProfile struct {
+
+	// ManagedAAD - AAD integration is managed by AKS.
+	// +optional
+	ManagedAAD *ManagedAAD `json:"managedAadProfile,omitempty"`
+
+	// DEPRECATED: use ManagedAAD instead
+	// LegacyAAD is old style azure active directory integration, user has to create the client app and server app.
+	// +optional
+	LegacyAAD *LegacyAAD `json:"legacyAadProfile,omitempty"`
+}
+
+// ManagedAAD - AAD integration is managed by AKS.
+type ManagedAAD struct {
+	// Managed - Whether to enable managed AAD.
+	// +kubebuilder:validation:Required
+	Managed *bool `json:"managed"`
+
+	// AdminGroupObjectIDs - AAD group object IDs that will have admin role of the cluster.
+	// +kubebuilder:validation:Required
+	AdminGroupObjectIDs *[]string `json:"adminGroupObjectIDs"`
+}
+
+// LegacyAAD is old style azure active directory integration, user has to create and manage the client app and server app for the AAD integration with AKS.
+type LegacyAAD struct {
+	// ClientAppID - The client AAD application ID.
+	// +kubebuilder:validation:Required
+	ClientAppID *string `json:"clientAppID"`
```

Comment:
any reason these are pointers if they are required, non-omitempty?

Reply:
In the beginning ing I was validating it a bit differently, so I needed it. I will change this to normal strings.

Reply:
I did not have legacyaad and managedaad, all the fields  where under one spec AADProfile. So for validation purpose i took them all as a pointer, but this was very tedious and not very clean. Latter I iterated to have legacy and managed as two separate specs. When I iterated i did not change it. But will change it now.